mint-client/src/api.rs

```diff
@@ -48,6 +49,30 @@ impl<'a> dyn FederationApi + 'a {
             }
         }
     }
+    pub async fn await_output_outcome<T: TryIntoOutcome + Send>(
+        &self,
+        outpoint: OutPoint,
+        timeout: Duration,
+    ) -> Result<T> {
+        let poll = || async {
+            let interval = Duration::from_secs(1);
```

Comment:
This should probably be configurable in the client?

Reply:
I'm not sure what you mean here.

1. Add to `HttpFederationApi`? But then the `await_output_outcome` won't know about it?
2. Add an `interval` parameter to `await_output_outcome`. and have callers pass that?

Reply:
Chat log:

one way would be to have a `await_output_outcome_enum` fn in the trait that uses `fetch_tx_outcome` internally

the enum variant would be implemented for `HttpFederationApi` and would have timeout access

while `await_output_outcome` would be generic and external and just call `await_output_outcome_enum` and do the conversion

Seems too much work rn, opening an issue instead.